charts/hub/portworx/templates/aws-prereqs/px-aws-prereqs-cm.yaml

```diff
@@ -0,0 +1,207 @@
+          else
+            unset RULE_EXISTS_SG
+            echo "Port TCP 20048 already allowed in security group $AWS_SG for $AWS_INSTANCE."
+          fi
+
+           # Check to see if security group already allows TCP 111
+          RULE_EXISTS_SG=$(aws ec2 describe-security-groups --region $AWS_REGION --filters Name=ip-permission.from-port,Values=111 Name=ip-permission.to-port,Values=111 Name=ip-permission.group-id,Values=$AWS_SG  --output text)
+          if [ -z "$RULE_EXISTS_SG" ]
+          then
+            unset RULE_EXISTS_SG
+            echo "Adding rule for TCP 111"
+            aws ec2 authorize-security-group-ingress --group-id $GROUP --protocol tcp --port 111 --region $AWS_REGION --source-group $GROUP
+          else
+            unset RULE_EXISTS_SG
+            echo "Port TCP 111 already allowed in security group $AWS_SG for $AWS_INSTANCE."
+          fi
+
+           # Check to see if security group already allows UDP 17002
+          RULE_EXISTS_SG=$(aws ec2 describe-security-groups --region $AWS_REGION --filters Name=ip-permission.from-port,Values=17002 Name=ip-permission.to-port,Values=111 Name=ip-permission.group-id,Values=$AWS_SG  --output text)
+          if [ -z "$RULE_EXISTS_SG" ]
+          then
+            unset RULE_EXISTS_SG
+            echo "Adding rule for UDP 17002"
+            aws ec2 authorize-security-group-ingress --group-id $GROUP --protocol udp --port 17002 --region $AWS_REGION --source-group $GROUP
+          else
+            unset RULE_EXISTS_SG
+            echo "Port UDP 17002 already allowed in security group $AWS_SG for $AWS_INSTANCE."
+          fi
+
+           # Check to see if security group already allows TCP 2049
+          RULE_EXISTS_SG=$(aws ec2 describe-security-groups --region $AWS_REGION --filters Name=ip-permission.from-port,Values=2049 Name=ip-permission.to-port,Values=2049 Name=ip-permission.group-id,Values=$AWS_SG  --output text)
+          if [ -z "$RULE_EXISTS_SG" ]
+          then
+            unset RULE_EXISTS_SG
+            echo "Adding rule for TCP 2049"
+            aws ec2 authorize-security-group-ingress --group-id $GROUP --protocol tcp --port 2049 --region $AWS_REGION --source-group $GROUP
+          else
+            unset RULE_EXISTS_SG
+            echo "Port UDP 2049 already allowed in security group $AWS_SG for $AWS_INSTANCE."
+          fi
+
+        else
+          echo "Could not find security group created by OpenShift installer, exiting."
+          exit 1
+        fi
+      echo ""
+      done
+
+      # Get the IAM instance profile for the next step
+      AWS_IAM_IP=$(aws ec2 describe-instances --instance-id $AWS_INSTANCE --region $AWS_REGION --query "Reservations[].Instances[].IamInstanceProfile[].Arn" --output text | sed 's|.*/||')
+    done
+
+    # Create the JSON for the necessary IAM permissions for Cloud Drives
+    cat << EOF >> /pattern-home/credentials/px-clouddrives.json

```

Comment:
(Just throwing it out there, not sure if it is entirely feasible in this context). It would be more gitops-like to use a credentials request https://github.com/validatedpatterns/common/blob/main/letsencrypt/templates/credentials-request.yaml . Ignore if it is not feasible here and/or more work than benefit